build-system/tasks/lint.js

```diff
@@ -124,7 +125,7 @@ function summarizeResults(results, fixedFiles) {
         )
       );
     }
-    process.exitCode = 1;
+    process.exitCode = errorCount || !isCiBuild() ? 1 : 0;
```

Comment:
Can you explain this change?

Reply:
Errors should always fail the build
Warnings (such as for the allowlisted imports) should not fail the build on CI
Chatted offline with @rsimha , looping in for approval on this file

Reply:
I didn't mean for this change to be made to `lint.js`. We want to continue failing CI builds when there are warnings.

Instead, you can change line 63 in `src/.eslintrc.js` as follows:

```js
'rules': {'import/no-restricted-paths': !isCiBuild() ? 1 : 0},
```

This will let you see warnings within `src` when you're doing local development, but not let those warnings show up during CI. Meanwhile, PRs that introduce unaccounted for warnings in other directories will be flagged during CI.

Reply:
That makes sense; will do

Reply:
Done! PTAL

Reply:
LGTM.

Reply:
Necromancing this conversation thread for a moment. With this configuration, branches touching any files with warnings get picked up as build targets, and `lint` emits warnings which in turn fails the task with a non-zero exit code. This means during local development, `amp pr-check` gets held up by the warnings and can't be useful on these branches. @rsimha do you have any suggestions to work around this? I'd like warnings to be displayed without failing the whole set of checks if possible. Could warnings not fail `lint` on non-CI builds? I don't see any lint rules that are warnings currently